src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,227 @@
+import React, { useContext, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+import { TodoContext } from '../../context/TodoContext';
+import * as todoService from '../../api/todos';
+import { Errors } from '../../types/Errors';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos, setErrorMessage, multiLoader, titleField } =
+    useContext(TodoContext);
+  const [deletedTodoId, setDeletedTodoId] = useState(0);
+  const [changedStatusTodo, setchangedStatusTodo] = useState(0);
+  const [localLoader, setLocalLoader] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+
+  const hasLoaderOnCreation = todo.id === 0;
+  const hasLoaderOnDeletion = localLoader && deletedTodoId === todo.id;
+  const hasLoaderOnCleaning = multiLoader && todo.completed;
+  const hasLoaderOnChangeStatus = localLoader && changedStatusTodo === todo.id;
+  const hasLoaderOnUpdating = localLoader && todo.title;
+  const isLoading =
+    hasLoaderOnCreation ||
+    hasLoaderOnDeletion ||
+    hasLoaderOnCleaning ||
+    hasLoaderOnChangeStatus ||
+    hasLoaderOnUpdating;
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+  }, [isEditing]);
+
+  const handleDoubleClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedTitle(event.target.value);
+  };
+
+  const removeTodo = () => {
+    setLocalLoader(true);
+    setDeletedTodoId(todo.id);
+    todoService
+      .deleteTodo(todo.id)
+      .then(() => {
+        setTodos(todos.filter(task => task.id !== todo.id));
+      })
+      .catch(() => {
+        setErrorMessage(Errors.DeleteError);
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      })
+      .finally(() => {
+        setLocalLoader(false);
+        setTimeout(() => {
+          if (titleField.current) {
+            titleField.current.focus();
+          }
+        }, 0);
```

Comment:
The same is here

Reply:
I don't know how do this. I don't know what to specify in the dependency array